pkg/build/strategies/sti/sti.go

```diff
@@ -169,14 +169,15 @@ func (b *STI) Prepare(request *api.Request) error {
 	}
 
 	// get the scripts
-	if request.ExternalRequiredScripts, err = b.installer.DownloadAndInstall(
-		b.requiredScripts, request.WorkingDir, true); err != nil {
+	required, err := b.installer.InstallRequired(b.requiredScripts, request.WorkingDir)
+	if err != nil {
 		return err
 	}
-
-	if request.ExternalOptionalScripts, err = b.installer.DownloadAndInstall(
-		b.optionalScripts, request.WorkingDir, false); err != nil {
-		glog.Warningf("Failed downloading optional scripts: %v", err)
+	optional := b.installer.InstallOptional(b.optionalScripts, request.WorkingDir)
+	if glog.V(1) {
```

Comment:
just a nit, I would put a newline here because the `if` is not related to the line above.